### DISCUSSION TITLE
How to do SQL queries in a Service?

### DISCUSSION THREAD

**Discussion by DiegelB** [oqtane/oqtane.framework#4564] [Q&A] [answered]
Hello, 

I am trying to use a service and interface that I have created in a module's client folder. The service is PlayerService and the interface is IPlayerService. Anytime the method in PlayerService is ran it tries to do a sql query to update the current users energy. When I run it keeps showing that my sql repository is null. I may be misunderstanding how to do sql calls as a service.

```
using System;
using Oqtane.Models;
using Oqtane.Modules;
using Oqtane.Repository;

namespace Oqtane.Services
{
    internal class PlayerService : ModuleBase
    {
        private readonly ISqlRepository _sqlRepository;
        public Tenant tenant;


        public void UpdatePlayerEnergy(UserStats player, Tenant tenant)
        {
            DateTime dateTime = DateTime.Now;
            TimeSpan timePassed = dateTime - player.LastTimeCheck;
            int minutesPassed = (int)timePassed.TotalMinutes;
            int energyToAdd = minutesPassed * 5;
            int currentEnergy = player.Energy;
            int newEnergy = currentEnergy + energyToAdd;
            int cappedEnergy = Math.Min(newEnergy, 100);


            if (minutesPassed < 1)
            {
                // do nothing
            }
            else
            {
// this is where I am getting the error
                 _sqlRepository.ExecuteNonQuery(tenant, $"UPDATE UserStats SET Energy = '100000' WHERE UserId = {player.UserId}");
                _sqlRepository.ExecuteNonQuery(tenant, $"UPDATE UserStats SET LastTimeCheck = '{dateTime}' WHERE UserId = {player.UserId}");
                player.LastTimeCheck = dateTime;
            }
        }
    }
}
```

**Comment by leigh-pointer** [comment]
@DiegelB  If you take a look at oqtane.framework\Oqtane.Client\Modules\Admin\Sql\Index.razor this should give you a really good starting point.

**Comment by sbwalker** [comment, marked as answer]
@DiegelB you are declaring a private ISqlRepository and never assigning it - so of course it will be null... you need to use dependency injection to get a reference to the actual ISqlRepository service. You are declaring a private Tenant variable and never initializing it. You should NOT use the Oqtane namespace - please use your own unique namespace for your module. You should not inherit from ModuleBase - this base class is only intended for razor components - not services. 

``
namespace YourName.Services 
{
    public class PlayerService : ITransientService
    {
        private readonly ITenantManager _tenantManager;
        private readonly ISqlRepository _sqlRepository;

        public PlayerService(ITenantManager tenantManager, ISqlRepository sqlRepository)
        {
            _tenantManager = tenantManager;
            _sqlRepository = sqlRepository;
        }

        public void UpdatePlayerEnergy(UserStats player)
        {
            _sqlRepository.ExecuteNonQuery(_tenantManager.GetTenant(), "your SQL query");
        }
    }
}        
```

**Reply by DiegelB** [reply]
Cool! That makes total sense. Thank you for authoring this framework and making it open source. Its my first time working with a client/server framework and I am learning so much. It really has reignited a lot of projects of mine that I gave up years ago. Sorry if my questions are a bit elementary. I'm getting there!
